app/lib/common/widgets/loading_overlay_screen.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:app/common/style/style.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+/// A factory for a loading overlay screen. Credit https://www.greycastle.se/loading-overlay-in-flutter/
+class LoadingOverlayScreen {
+  BuildContext _context;
+
+  LoadingOverlayScreen._create(this._context);
+
+  factory LoadingOverlayScreen.of(BuildContext context) {
+    return LoadingOverlayScreen._create(context);
+  }
+
+  void hide() {
+    Navigator.of(_context, rootNavigator: true).pop();
+  }
+
+  void show() {
+    showDialog(
+        context: _context,
+        barrierDismissible: false,
+        child: WillPopScope(
+            child: _LoadingOverlayScreen(), onWillPop: () async => false));
```

Comment:
[Interesting way of providing onWillPop](https://stackoverflow.com/questions/53156666/detect-back-button-press-while-dialog-is-open-in-flutter)